### DISCUSSION TITLE
How to place point source?

### DISCUSSION THREAD

**Discussion by HSCmi** [thliebig/openEMS-Project#46] [Q&A Usage] [answered]
Hi,

I am very new to using openEMS (and also new to FDTD simulations), and would like to ask a very basic question.

I would like to do a simulation on metasurface with point source as incident light.
My question is: How can I place a point source?

I am very sorry to ask a very basic question, but I hope someone could help me with this problem.

**Comment by thliebig** [comment]
Well there is no such thing in the real world like a point source... You will need to think more about how you intent to excite your surface. E.g. which polarization and wave directions etc.. Does a simple antenna do it, e.g. a dipole? Or maybe a plane wave excite with all of its very own difficulties?

**Reply by HSCmi** [reply]
Thank you for your response. And I apologize for the confusion.

Like you have suggested, I would like to create a dipole source.
The abstract of the model I would like to create is like this:
![image](https://user-images.githubusercontent.com/128348785/226839473-4013b63e-897d-45b5-85bd-a57f1a124556.png)

The reason why I need to create dipole source (+ plane wave) is because I would like to do an optimization of metasurface (with functionality of lens)  using adjoint method.

**Comment by 0xCoto** [comment, marked as answer]
You're probably looking for a plane wave source if you're looking to simulate metasurfaces.

**Reply by alexpapad95** [reply]
How can I add the plane wave source and what are the respective difficulties?

**Reply by HSCmi** [reply]
@0xCoto 
Thank you for your response.
In my case, I would like to use plane wave source + dipole source.
This is because I would like to do an optimization of metasurface (with functionality of lens) using adjoint method.

**Reply by 0xCoto** [reply]
I haven't worked with plane waves in openEMS (only other software), but it seems to be fairly straightforward:

https://github.com/thliebig/openEMS/blob/master/matlab/examples/other/PlaneWave.m

https://youtu.be/NbKZjPlcQKE

For an omnidirectional source, your best bet would probably be to model 2 cylinders and form a dipole antenna.

I'm not sure whether reducing the dimensions of the plane wave source could help produce more of a spherical wave or not. Another idea would be to have a plane wave source with a narrow slit, causing diffraction (emulating a spherical wave emerging from the slit).
